docs/spec/efficient_querying.rst

```diff
@@ -0,0 +1,98 @@
+.. _efficient_querying:
+
+==================
+Efficient Querying
+==================
+
+In general, Kartothek should know as early as possible which data to exclude, so
+Kartothek doesn't have to read and transfer it. If you're going to query a
+dataset a lot, you should consider storing it in an optimized layout. See
+:ref:`writing`.
+
+
+.. _querying_process:
+
+Background
+----------
+
+To understand how to query efficiently, you should understand Kartothek's
+querying process:
+
+When querying data, Kartothek tries to apply the given predicates as early as
+possible, so it has to read and parse as few parquet files as possible:
+
+- Kartothek looks if there exists an index for the filtered column. This may
```

Comment:
We may want to add keywords like "partition pruning" to help already advanced users to understand what's going on. This is a rather common terminology for this process, e.g. drill has its own section for it (we don't need that) https://drill.apache.org/docs/partition-pruning-introduction/